### ISSUE TITLE
a11y: Add aria labels for header action icon buttons

### ISSUE THREAD

**Issue by edenhernandez-suse** [rancher/dashboard#8113]
a11y Issue: 26

## Context
The header actions use icon buttons without any text description:

![image](https://user-images.githubusercontent.com/120640736/217539604-ccb0698c-6a79-4775-a0bf-487e14427df3.png)

## The bug
They have a tooltip that allows mouse and keyboard users to better understand their meaning.
For screen-readers, however, they are read as "_button_".

## Expected Result
These buttons should have an aria-label with the same text shown in the tooltip.

Part of rancher/dashboard#7968
